### ISSUE TITLE
No desktop-style list box widget with keyboard navigation that keeps track of which item is selected and has proper screen reader support

### ISSUE THREAD

**Issue by emassey0135** [flutter/flutter#104362]
## Use case

First, I would like to display information in a list in a way that allows screen reader users to easily read the information on desktop platforms, preferably using standard platform conventions. I tried using a Flutter ListView widget with items as Text widgets with each Text widget wrapped in a Focus widget. I could navigate the list by pressing Tab and Shift+Tab on Windows and Linux, but the arrow keys behaved inconsistently, screen readers did not announce that this was a list, and I could not easily leave the list to move to other controls. Secondly, I would like to create a list that allows the user to select an item by moving to it with the arrow keys or with a mouse click, and change other parts of the window based on the current selection, perhaps by displaying information about the selected item in another part of the window. I would also like an easy way to move focus away from the list with the keyboard, so Tab and Shift+Tab should navigate away from the list and not change the selected item.

## Proposal

A new widget could be implemented that appears as a single widget to the focus traversal system, mostly looks like a ListView widget visually, has state that stores which of its children is selected, lets the user change the selection with the up and down arrow keys, and has semantic information that informs screen readers that it is a list, its label, the number of items it has, and which item is currently selected. There could also be a widget that does this for a grid, by letting the user use the left and right arrow keys as well to navegate between columns. Examples of native widgets that implement this are [List Box](https://docs.microsoft.com/en-us/windows/win32/controls/list-boxes) and [List View](https://docs.microsoft.com/en-us/windows/win32/controls/list-view-control-reference) on Win32, [GtkListBox](https://docs.gtk.org/gtk3/class.ListBox.html) on GTK 3, and perhaps [NSTableView](https://developer.apple.com/documentation/appkit/nstableview) on Cocoa.

**Comment by darshankawar** [comment]
@emassey0135 
Thanks for the report. Implementing widgets specifically tailored for windows + accessibility may not be a feasible solution / proposal. Have you looked at https://pub.dev/packages/fluent_ui and https://pub.dev/packages/flutter_acrylic plugins that implements windows style design and check if it helps to achieve your use case ?

**Comment by emassey0135** [comment]
The flutter_acrylic package only changes window styles and does not have widgets, and the fluent_ui package has a drop down button widget and a combo box widget, but nothing that appears like a ListView widget and has the behaviour I am looking for. The drop down button and combo box are not suitable because they do not display all their items at once and must be expanded to see the items. It has a navigation pane that seems promising, but not a general list box widget. It has a list tile that says it is meant to be used with ListView, which seems to refer to the one in the material package. Should I report an issue on the fluent_ui package?

**Comment by darshankawar** [comment]
you may try that and see what the plugin owner comes up with. Also, you may look for similar community packages on pub.dev to check if any of them helps to address your proposal.

For now, I am going ahead and closing it. If you disagree, write in comments and I'll reopen it.
